src/components/input/input.e2e.ts

```diff
@@ -76,7 +76,78 @@ describe("calcite-input", () => {
       }
     ]));
 
-  it("can be disabled", () => disabled("calcite-input"));
+  describe("calcite-input when disabled", () => {
+    it("can be disabled", () => disabled("calcite-input"));
+  });
+
+  describe("calcite-input when readOnly", () => {
+    it("sets internals to readOnly or disabled when readOnly is true", async () => {
```

Comment:
Prefer testing behavior and not the internals. This will prevent the test from breaking if we ever refactored it. Since you already do this in the following tests, I think this one can be removed.